### DISCUSSION TITLE
Facets sort alphabetically?

### DISCUSSION THREAD

**Discussion by oscarchic** [simonw/datasette#1908] [General] [answered]
Is possible to sort facets alphabetically?

Thanks in advance,
Oscar

**Comment by simonw** [comment, marked as answer]
It's not at the moment, but this is an interesting feature idea.

Right now you can achieve this using a custom SQL query, for example for the list of states on https://congress-legislators.datasettes.com/legislators/legislator_terms?_facet=state&_facet_size=max

You can run this:

https://congress-legislators.datasettes.com/legislators?sql=select+state%2C+count%28*%29+from+legislator_terms+group+by+state+order+by+state

```sql
select
  state,
  count(*)
from
  legislator_terms
group by
  state
order by
  state
```

Or if you want to be able to click a link to view the results by state:
```sql
select
  state,
  count(*),
  'https://congress-legislators.datasettes.com/legislators/legislator_terms?state=' || state as url
from
  legislator_terms
group by
  state
order by
  state
```
https://congress-legislators.datasettes.com/legislators?sql=select%0D%0A++state%2C%0D%0A++count%28*%29%2C%0D%0A++%27https%3A%2F%2Fcongress-legislators.datasettes.com%2Flegislators%2Flegislator_terms%3Fstate%3D%27+%7C%7C+state+as+url%0D%0Afrom%0D%0A++legislator_terms%0D%0Agroup+by%0D%0A++state%0D%0Aorder+by%0D%0A++state&p0=PA

You could even write a custom faceting plugin that does this, but it would be tricky as there aren't any good examples of faceting plugins at the moment: https://docs.datasette.io/en/stable/plugin_hooks.html#register-facet-classes

I've opened an issue here to think about how this could work in Datasette itself:
- https://github.com/simonw/datasette/issues/1909

**Comment by oscarchic** [comment]
Hi Simon, 

Fantastic!! Thanks a lot for the answer. 
I try to understand ... and use it.

Best,
Òscar
